### ISSUE TITLE
Extreme lag on 1.19.2 

### ISSUE THREAD

**Issue by warplord** [CookieCNetwork/Chipped#176]
### Bug Description

I installed chipped today to my modpack and I got some extreme lag, even crashed because server couldn't keep up. Removed the mod - lag was gone. 

### How to Reproduce?

Just instal the mod. 

### Expected Behavior

no lag

### Version

chipped-forge-1.19.2-2.1.0.jar

### Mod Loader Version

forge 43.2.4

### Mod Loader

Forge

### Logs or additional context

No logs - since the game just lags, i saw my ram usage was at 99% with the mod, at i allocated 9GB to the modpack.

**Comment by sargeow** [comment]
[Quilt 1.19.2] [Loader 18.2]

Also having performance issues, installed chipped in my 200+ modpack and my Servers RAM and CPU usage sky rocketed and the TPS would sit at silly numbers with it getting worse by the second. managed to max the server out at 12g's of ram VS its usual 5-8g. im sure this is an issue with java just being bad but who knows, might just be a bug.

**Comment by 3dguru** [comment]
Also having the same issue, seems chipped - fabric  has a big memory leak issue  using between 3 -6 gigs of ram either single player or server side.

**Comment by aaronhowser1** [comment]
With Chipped, making a new world takes upwards of 3 minutes. 2 of those minutes were at 0-1% loaded. Removing it brings it down to 1 minute.

chipped-forge-1.19.2-2.1.1

**Comment by AlexNijjar** [comment]
does it still happen if you install lazy dfu? https://www.curseforge.com/minecraft/mc-mods/lazydfu

**Comment by sargeow** [comment]
> does it still happen if you install lazy dfu? https://www.curseforge.com/minecraft/mc-mods/lazydfu

LazyDFU didn't fix my issue. Client side on a new world seems to be fine however on my pre-existing server world the zombies moon walk. There might be something of use in spark logs here. i can try using a new server side world to see if the issue persists. (note using 200+ mods)

https://spark.lucko.me/E4j8OyOaPp
![spark tps](https://user-images.githubusercontent.com/89289084/222778727-298681d6-1ad2-4ba0-a795-42122bcdf670.png)

**Comment by ThatGravyBoat** [comment]
When you mean pre-existing server do you mean a 1.18.2 world on the server? or do you mean a server you just added chipped to?

**Comment by AlexNijjar** [comment]
can you run /perf start on your server and send the output zip?

**Comment by sargeow** [comment]
> When you mean pre-existing server do you mean a 1.18.2 world on the server? or do you mean a server you just added chipped to?

The latter.

**Comment by gregest** [comment]
Fabric 1.19.2 chipped-fabric-1.19.2-2.1.1.jar

Can confirm it also happened to me.   

I installed the mod in a modpack, cpu usage skyrocketed even in the main menu, the ram garbage collector was spinning through numbers at an insane speed and I had like 2fps, removing the mod fixes it.   


https://user-images.githubusercontent.com/56032903/223547777-b352682e-5d9f-4066-a356-55cc5c41b24e.mp4

**Comment by 1oglop1** [comment]
I can confirm the same problem: 
My original world is 1.19.2 with chipped 2.0.1
After update to chipped 2.1.0 + resourcefullib 1.1.21 - game became unplayable

edit: 
@AlexNijjar I had a default Mem allocation of 2GB and Chipped caused extreme lag, then I increased mem to 4GB and it was better. 

I did some testing in 2 worlds, 1 flat, and 2 normal. Both worlds do not contain any chipped blocks and were working fine with 2.0.1 (except the crash with REI). 

With chipped 2.1.2, memory was jumping between 2GB - 3.5GB. Without Chipped it stayed around 2GB.

here are mods without Chipped 1.9.2-2.1.2

```
	- animal_feeding_trough 1.0.3+1.19.2
	- appleskin 2.4.1+mc1.19
	- architectury 6.3.49
	- betteradvancements 0.2.2.142
	- byg 2.0.1.1
	- caffeineconfig 1.0.0+1.17 via sodium-extra
	- cloth-basic-math 0.6.1 via cloth-config
	- cloth-config 8.0.75
	- clumps 9.0.0+16
	- com_electronwill_night-config_core 3.6.6 via byg
	- com_electronwill_night-config_toml 3.6.6 via byg
	- com_eliotlash_mclib_mclib 20 via geckolib3
	- com_github_llamalad7_mixinextras 0.1.1 via porting_lib
	- com_google_code_findbugs_jsr305 3.0.2 via create
	- com_logisticscraft_occlusionculling 0.0.6-SNAPSHOT via entityculling
	- corgilib 1.0.0.32
	- create 0.5.0.i-961+1.19.2
	- ctm 1.0.0+1.19
	- decorative_blocks 3.0.0
	- disable_custom_worlds_advice 3.0
	- dripstone_fluid_lib 2.0.1 via milk
	- dungeons_arise 2.1.54
	- entityculling 1.5.2-mc1.19
	- error_notifier 1.0.9 via roughlyenoughitems
	- fabric-api 0.75.1+1.19.2
	- fabric-api-base 0.4.15+8f4e8eb390 via fabric-api
	- fabric-api-lookup-api-v1 1.6.14+93d8cb8290 via fabric-api
	- fabric-biome-api-v1 9.1.0+82db024a90 via fabric-api
	- fabric-block-api-v1 1.0.2+e415d50e90 via fabric-api
	- fabric-blockrenderlayer-v1 1.1.25+cafc6e8e90 via fabric-api
	- fabric-client-tags-api-v1 1.0.5+b35fea8390 via fabric-api
	- fabric-command-api-v1 1.2.16+f71b366f90 via fabric-api
	- fabric-command-api-v2 2.2.1+413cbbc790 via fabric-api
	- fabric-commands-v0 0.2.33+df3654b390 via fabric-api
	- fabric-containers-v0 0.1.41+df3654b390 via fabric-api
	- fabric-content-registries-v0 3.5.1+7c6cd14d90 via fabric-api
	- fabric-convention-tags-v1 1.2.2+27265b3490 via fabric-api
	- fabric-crash-report-info-v1 0.2.8+aeb40ebe90 via fabric-api
	- fabric-data-generation-api-v1 5.3.7+413cbbc790 via fabric-api
	- fabric-dimensions-v1 2.1.35+0d0f210290 via fabric-api
	- fabric-entity-events-v1 1.5.4+9244241690 via fabric-api
	- fabric-events-interaction-v0 0.4.34+562bff6e90 via fabric-api
	- fabric-events-lifecycle-v0 0.2.35+df3654b390 via fabric-api
	- fabric-game-rule-api-v1 1.0.24+b6b6abb490 via fabric-api
	- fabric-item-api-v1 1.6.5+dce4603e90 via fabric-api
	- fabric-item-groups-v0 0.3.38+9244241690 via fabric-api
	- fabric-key-binding-api-v1 1.0.25+5c4fce2890 via fabric-api
	- fabric-keybindings-v0 0.2.23+df3654b390 via fabric-api
	- fabric-language-kotlin 1.8.3+kotlin.1.7.10
	- fabric-lifecycle-events-v1 2.2.4+1b46dc7890 via fabric-api
	- fabric-loot-api-v2 1.1.12+83a8659290 via fabric-api
	- fabric-loot-tables-v1 1.1.15+9e7660c690 via fabric-api
	- fabric-message-api-v1 5.0.7+93d8cb8290 via fabric-api
	- fabric-mining-level-api-v1 2.1.23+33fbc73890 via fabric-api
	- fabric-models-v0 0.3.21+c6af733c90 via fabric-api
	- fabric-networking-api-v1 1.2.11+10eb22f490 via fabric-api
	- fabric-networking-v0 0.3.28+df3654b390 via fabric-api
	- fabric-object-builder-api-v1 4.2.1+d8ef690890 via fabric-api
	- fabric-particles-v1 1.0.14+4d0d570390 via fabric-api
	- fabric-recipe-api-v1 1.0.1+413cbbc790 via fabric-api
	- fabric-registry-sync-v0 0.9.32+9244241690 via fabric-api
	- fabric-renderer-api-v1 1.2.1+1adbf27790 via fabric-api
	- fabric-renderer-indigo 0.8.0+1adbf27790 via fabric-api
	- fabric-renderer-registries-v1 3.2.24+df3654b390 via fabric-api
	- fabric-rendering-data-attachment-v1 0.3.19+6e0787e690 via fabric-api
	- fabric-rendering-fluids-v1 3.0.11+4d0d570390 via fabric-api
	- fabric-rendering-v0 1.1.27+df3654b390 via fabric-api
	- fabric-rendering-v1 1.12.1+d8ef690890 via fabric-api
	- fabric-resource-conditions-api-v1 2.1.2+aae9039d90 via fabric-api
	- fabric-resource-loader-v0 0.8.3+251d254a90 via fabric-api
	- fabric-screen-api-v1 1.0.32+4d0d570390 via fabric-api
	- fabric-screen-handler-api-v1 1.3.7+1cc24b1b90 via fabric-api
	- fabric-sound-api-v1 1.0.2+c4f28df590 via fabric-api
	- fabric-textures-v0 1.0.24+aeb40ebe90 via fabric-api
	- fabric-transfer-api-v1 2.1.6+413cbbc790 via fabric-api
	- fabric-transitive-access-wideners-v1 1.3.3+08b73de490 via fabric-api
	- fabricloader 0.14.13
	- flywheel 0.6.8.a-4 via create
	- forge_tags 3.0 via porting_lib
	- forgeconfigapiport 4.2.9 via create
	- geckolib3 3.1.40
	- gravestones v1.13
	- indium 1.0.9+mc1.19.2
	- inventoryprofilesnext 1.9.2
	- java 17
	- lambdynlights 2.1.2+1.19
	- libipn 2.0.2
	- lithium 0.10.2
	- malilib 0.13.0
	- milk 1.0.51 via create
	- minecraft 1.19.2
	- minihud 0.23.3
	- mm 2.3 via porting_lib
	- modmenu 4.1.0
	- org_jetbrains_kotlin_kotlin-reflect 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk7 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk8 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlinx_atomicfu-jvm 0.18.3 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.6.4 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.6.4 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-datetime-jvm 0.4.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm 1.4.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.4.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.4.0 via fabric-language-kotlin
	- org_joml_joml 1.10.4 via sodium
	- phosphor 0.8.1
	- porting_lib 2.1.784+1.19.2 via create
	- porting_lib_accessors 2.1.784+1.19.2 via porting_lib
	- porting_lib_attributes 2.1.784+1.19.2 via porting_lib
	- porting_lib_base 2.1.784+1.19.2 via porting_lib
	- porting_lib_common 2.1.784+1.19.2 via porting_lib
	- porting_lib_constants 2.1.784+1.19.2 via porting_lib
	- porting_lib_entity 2.1.784+1.19.2 via porting_lib
	- porting_lib_extensions 2.1.784+1.19.2 via porting_lib
	- porting_lib_fake_players 2.1.784+1.19.2 via porting_lib
	- porting_lib_lazy_registration 2.1.784+1.19.2 via porting_lib
	- porting_lib_model_generators 2.1.784+1.19.2 via porting_lib
	- porting_lib_model_loader 2.1.784+1.19.2 via porting_lib
	- porting_lib_models 2.1.784+1.19.2 via porting_lib
	- porting_lib_networking 2.1.784+1.19.2 via porting_lib
	- porting_lib_obj_loader 2.1.784+1.19.2 via porting_lib
	- porting_lib_transfer 2.1.784+1.19.2 via porting_lib
	- pride 1.1.2+1.19 via lambdynlights
	- reach-entity-attributes 2.3.0 via create
	- reeses-sodium-options 1.4.9+mc1.19.2-build.67
	- registrate-fabric 1.1.50-MC1.19.2 via create
	- resourcefullib 1.1.23
	- roughlyenoughitems 9.1.572
	- serialization_hooks 0.3.26 via porting_lib
	- shiftscrollfix 1.0.1
	- sodium 0.4.4+build.18
	- sodium-extra 0.4.16+mc1.19.2-build.90
	- spruceui 4.0.0+1.19 via lambdynlights
	- team_reborn_energy 2.2.0 via theoneprobe
	- terrablender 2.0.1.129
	- theoneprobe 1.19-6.2.2
	- tweakeroo 0.14.3
	- wi_zoom 1.4-MC1.19.2
```

**Comment by AlexNijjar** [comment]
> With chipped 2.1.2, memory was jumping between 2GB - 3.5GB. Without Chipped it stayed around 2GB.

is there still extreme lag on  2.1.2 or is it just higher memory usage?

**Comment by 1oglop1** [comment]
> > With chipped 2.1.2, memory was jumping between 2GB - 3.5GB. Without Chipped it stayed around 2GB.
> 
> is there still extreme lag on 2.1.2 or is it just higher memory usage?

I think that overall 2.1.2 is a significant improvement over 2.1.0.
But memory usage of 2.1.2 is at least 600-700MB more than 2.0.1. 

Loading: 
With 2.1.0 the world loading took way longer, however, with 2.1.2 there is a still noticeable delay when compared to 2.0.1. 

In-game performance:
My flat world with very few blocks and 0 chipped blocks did not show performance problems. 
Loading the normal world was quite problematic at the beginning. It took quite a lot of time to place the player on the block and render everything (that state where you are falling into the void). 
After loading everything I noticed a significant lag every `N` seconds, which made me look at the memory consumption. 
I noticed the memory was close to full 2GB. 

And finally increasing the memory to 4GB helped, the only thing left is the initial loading takes longer.

Is there anything I can to do help debug this better?

**Comment by ThatGravyBoat** [comment]
The extra initial load time is because of datafixers which were a highly requested thing by a multitude of people, installing the mod lazydfu or enabling the JVM option `-Dchipped.datafixers=false` should turn them off until they are needed or should disable them entirely.

We calculated how long it takes with chipped to load it takes between 2-3 minutes this time is all because of how slow Minecrafts model loading is.

**Comment by t-Ghost-t** [comment]
Same issue here, chipped causes my server to lag (maxes out 16gb ram usage on a server that idles at 4gb and peaks at 9gb without it) until disconnects or crashes occur. Impossible to use.

**Comment by AlexNijjar** [comment]
This should be fixed in 1.20

**Comment by Dreaming-Codes** [comment]
> This should be fixed in 1.20

What's the commit that fixed the issue so that I can backport it to 1.19.2 for my server

**Comment by xinwolfthorn** [comment]
i will say that this problem persists in 1.21.1..
